docs/en/observability/monitor-gcp.asciidoc

```diff
@@ -0,0 +1,457 @@
+      "cluster": ["monitor", "read_ilm"],
+      "index": [
+        {
+          "names": ["metricbeat-*"],
+          "privileges": ["view_index_metadata", "create_doc"]
+        }
+      ]
+    }
+  }
+}
+----
+
+. The response contains an `api_key` and an `id` field, which can be stored in
+the {metricbeat} keystore in the following format: `id:api_key`.
++
+[source,bash]
+----
+echo -n "IhrJJHMB4JmIUAPLuM35:1GbfxhkMT8COBB4JWY3pvQ" | ./metricbeat keystore add ES_API_KEY --stdin
+----
++
+[NOTE]
+=====
+Make sure you specify the `-n` parameter; otherwise, you will have
+painful debugging sessions, due to adding a newline at the end of
+your API key.
+=====
+
+. To see if both settings have been stored, run `./metricbeat keystore list`.
+
+. Then, configure {metricbeat} to output to {ess}. Edit the `metricbeat.yml`
+configuration file and add the following lines to the end of the file.
++
+[source,yml]
+----
+cloud.id: ${CLOUD_ID}
+output.elasticsearch:
+  api_key: ${ES_API_KEY}
+----
+
+. Finally, test if the configuration is working. If it is not working,
+verify if you used the right credentials and add them again.
++
+[source,bash]
+----
+./metricbeat test output
+----
+
+Now that the output is working you are going to setup the input (GCP).
+
+[discrete]
+=== Step 3: Configure {metricbeat} Google Cloud Platform module
+
+In order to collect metrics from Google Cloud Platform, use the
+https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-gcp.html[Google Cloud Platform]
+module. This module periodically fetches monitoring metrics from Google Cloud
+Platform using
+https://cloud.google.com/monitoring/api/metrics_gcp[Stackdriver Monitoring API]
+for Google Cloud Platform services.
+
+[WARNING]
+====
+Extra GCP charges on Stackdriver Monitoring API requests may be generated by
+this module. Please see
+https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-gcp.html#gcp-api-requests[rough estimation of the number of API calls]
+for more details.
+====
+
+. First, enable the GCP module.
++
+[source,bash]
+----
+./metricbeat modules enable gcp
+----
+
+. Next, edit the `modules.d/gcp.yml` file to configure which metrics to
+collect.
++
+[source,yml]
+----
+- module: googlecloud
+  metricsets:
+    - compute <1>
+  zone: "" <2>
+  project_id: "elastic-education" <3>
+  period: 1m <4>
+  credentials_file_path: "/home/ubuntu/credentials.json" <5>
+----
++
+<1> The `compute` metricset is a predefined metricset that collects some
+GCP compute metrics.
+<2> Defines which zones to monitor, an empty value collects data from *all* zones
+<3> Collects metrics within the `elastic-education` project-id.
+<4> Collects metrics every minute
+<5> The GCP credential file that you generated earlier. (Don't forget to create
+the file if it does not exist and to use the correct full path).
+
+. Now, test the input to check if {metricbeat} can collect data.
+[source,bash]
++
+----
+./metricbeat test modules gcp
```

Comment:
Mention what a good/bad response here looks like, so the user knows, that the setup worked?